ai2thor/video_controller.py

```diff
@@ -27,28 +27,18 @@ def __init__(
         cam_fov=60,
         **controller_kwargs,
     ):
-        super().__init__(continuous=True, **controller_kwargs)
-        self.step(
-            action="AddThirdPartyCamera",
-            rotation=initial_camera_rotation,
-            position=initial_camera_position,
-            fieldOfView=initial_camera_fov,
-        )
-
         self.saved_frames = []
         self.ceiling_off = False
         self.initial_cam_rot = cam_rot.copy()
         self.initial_cam_pos = cam_pos.copy()
         self.initial_cam_fov = cam_fov
```

Comment:
I've reproduced the bug. To fix it, I think all that need to be changed is:

```python
        self.step(
            action="AddThirdPartyCamera",
            rotation=initial_camera_rotation,
            position=initial_camera_position,
            fieldOfView=initial_camera_fov,
        )

        self.saved_frames = []
        self.ceiling_off = False
        self.initial_cam_rot = cam_rot.copy()
        self.initial_cam_pos = cam_pos.copy()
        self.initial_cam_fov = cam_fov
```
becomes
```python
        self.saved_frames = []
        self.ceiling_off = False
        self.initial_cam_rot = cam_rot.copy()
        self.initial_cam_pos = cam_pos.copy()
        self.initial_cam_fov = cam_fov

        self.step(
            action="AddThirdPartyCamera",
            rotation=self.initial_cam_rot,
            position=self.initial_cam_pos,
            fieldOfView=self.initial_cam_fov,
        )
```

Reply:
This still doens't work for me:

```
Traceback (most recent call last):
  File "video_controller_new.py", line 391, in <module>
    with VideoController() as vc:
  File "video_controller_new.py", line 30, in __init__
    super().__init__(continuous=True, **controller_kwargs)
  File "ai2thor/controller.py", line 535, in __init__
    event = self.reset(scene)
  File "video_controller_new.py", line 50, in reset
    rotation=self.initial_cam_rot,
AttributeError: 'VideoController' object has no attribute 'initial_cam_rot'
```

Reply:
To clarify, did you move:
```python
        self.step(
            action="AddThirdPartyCamera",
            rotation=initial_camera_rotation,
            position=initial_camera_position,
            fieldOfView=initial_camera_fov,
        )
```
below:
```python
        self.saved_frames = []
        self.ceiling_off = False
        self.initial_cam_rot = cam_rot.copy()
        self.initial_cam_pos = cam_pos.copy()
        self.initial_cam_fov = cam_fov
```
If so, it should be setting `self.initial_cam_rot` directly prior to using it.

Reply:
Yes I did. :) I think the problem is that the VideoController and Controller objects's reset function are going back and forth and self.<object property> is defined in one class's scope and not in the other one. There's a number of ways to resolve it but you'd know better given the many uses you want to support 

```
class VideoController(Controller):
    def __init__(
        self,
        cam_rot=dict(x=85, y=225, z=0),
        cam_pos=dict(x=-1.25, y=7.0, z=-1.0),
        cam_fov=60,
        **controller_kwargs,
    ):
        super().__init__(continuous=True, **controller_kwargs)

        self.saved_frames = []
        self.ceiling_off = False
        self.initial_cam_rot = cam_rot.copy()
        self.initial_cam_pos = cam_pos.copy()
        self.initial_cam_fov = cam_fov

        self.step(
            action="AddThirdPartyCamera",
            rotation=self.initial_cam_rot,
            position=self.initial_cam_pos,
            fieldOfView=self.initial_cam_fov,
        )

    def reset(self, scene=None, **init_params):
        """Changes the scene and adds a new third part
    ...
```